server/messaging/reporting.py

```
@@ -70,9 +70,12 @@ def get_notifications (db):
 
     if 'new_devices' in sections:
         # Compose New Devices Section (no empty lines in SQL queries!)
+        # Note: NTFPRCS_new_dev_condition should be validated/sanitized at the settings level
+        # to prevent SQL injection. For now, we preserve existing functionality but flag the risk.
+        new_dev_condition = get_setting_value('NTFPRCS_new_dev_condition').replace('{s-quote}',"'")
         sqlQuery = f"""SELECT eve_MAC as MAC, eve_DateTime as Datetime, devLastIP as IP, eve_EventType as "Event Type", devName as "Device name", devComments as Comments  FROM Events_Devices
                         WHERE eve_PendingAlertEmail = 1
-                        AND eve_EventType = 'New Device' {get_setting_value('NTFPRCS_new_dev_condition').replace('{s-quote}',"'")} 
+                        AND eve_EventType = 'New Device' {new_dev_condition} 
                         ORDER BY eve_DateTime"""   
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)